tests/services/test_file_workflow_service.py

```
@@ -310,13 +308,25 @@ def test_process_single_file(
-    mock_select_location.return_value = MagicMock(id=1, path=str(cold_path))
-    mock_move.return_value = (True, None, "checksum_after")
-    mock_checksum.return_value = "checksum_before"
+    mock_select_location.return_value = MagicMock(
+        id=1,
+        path=str(cold_path),
+        backend_type=ColdStorageBackendType.LOCAL,
+        is_encrypted=False,
+        operation_mode=OperationType.MOVE,
+    )
+
+    def freeze_side_effect(file, monitored_path, storage_location, pin, db, initiated_by):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `freeze_side_effect`

(ANN202)